### ISSUE TITLE
Issue with RTT measurement

### ISSUE THREAD

**Issue by soora-jp** [Unipisa/Simu5G#122]
Hello.

I'm measuring the end-to-end round trip time (RTT) between UE and MEC host in MEC environment. Most UEs' RTTs are between 13-14ms, but specific UEs' RTTs are between 19-20ms. It looks like it takes 2-times transmissions to send the message because of not enough sduSize.
I have no idea why this happens.
Could you give me some insights about it?

Application specification
- the size of request and response messages is fixed with 100Byte
- intervals between sending messages are random within 1s
- transport layer is TCP

Thank you and Best regards.

![rtt](https://user-images.githubusercontent.com/28481373/224246693-a8cf9678-5b7b-436d-a5fd-ee466faa8266.svg)

**Comment by soora-jp** [comment]
Thank you for your response.

As you said, `averageCqiUl` of UEs with higher RTT were 14 while `averageCqiUl` of UEs with lower RTT were 15. 
However, in my scenario, all UEs are in exactly the same position.
I tried to find the reason for the low CQI by finding where the CQI was calculated, but I couldn't.

So, I would like to know the following
1. Where is the CQI calculated?
2. What caused the low CQI other than the UEs’ position?

Best regards.
Sora

**Comment by giovanninardini** [comment]
CQI is calculated by the function _requestFeedback()_, in classes _LtePhyEnb_ and _LtePhyEnbD2D_.
One possible cause for different CQIs for UEs in the same position may be due to, for example, fading and/or shadowing effects (if enabled in the channel model).

**Comment by Ali-K-Abbas** [comment]
@soora-jp Dear, Good day 
Can You help me with how you exactly measuring the end-to-end round trip time (RTT) between UE and MEC host in MEC environment?

**Comment by soora-jp** [comment]
@giovanninardini 
Hello. I sincerely apologize for not replying. I will learn wireless communication and 5G protocols from scratch again. Thank you.

**Comment by soora-jp** [comment]
@Ali-K-Abbas 
Hello. I’m sorry for the late response. 

When I started my current work, there was no MEC application to measure e2e RTT, so I implemented it myself. However, in version 1.2.1 of simu5g, there is [mecRequestResponseApp](https://github.com/Unipisa/Simu5G/tree/master/src/apps/mecRequestResponseApp), and it looks enabled to measure e2e RTT. So I recommend using it. For reference, I used [tag function of INET](https://inet.omnetpp.org/docs/developers-guide/ch-tags.html) to embed request time in packets.
